phoenix-ios/phoenix-ios/views/SendView.swift

```diff
@@ -733,7 +737,252 @@ struct SendingView: View {
 			NSLocalizedString("Sending payment", comment: "Navigation bar title"),
 			displayMode: .inline
 		)
-		.zIndex(0) // [SendingView, ValidateView, ScanView]
+		.zIndex(0) // [SendingView, ValidateView, LoginView, ScanView]
+	}
+}
+
+struct LoginView: View, ViewName {
+	
+	@ObservedObject var mvi: MVIState<Scan.Model, Scan.Intent>
+	
+	enum MaxImageHeight: Preference {}
+	let maxImageHeightReader = GeometryPreferenceReader(
+		key: AppendValue<MaxImageHeight>.self,
+		value: { [$0.size.height] }
+	)
+	@State var maxImageHeight: CGFloat? = nil
+	
+	@ViewBuilder
+	var body: some View {
+		
+		ZStack {
+		
+			if AppDelegate.showTestnetBackground {
+				Image("testnet_bg")
+					.resizable(resizingMode: .tile)
+			}
+			
+			// I want the height of these 2 components to match exactly:
+			// Button("<img> Login")
+			// HStack("<img> Logged In")
+			//
+			// To accomplish this, I need the images to be same height.
+			// But they're not - unless we measure them, and enforce matching heights.
+			
+			Image(systemName: "bolt")
+				.imageScale(.large)
+				.font(.title2)
+				.foregroundColor(.clear)
+				.read(maxImageHeightReader)
+			
+			Image(systemName: "hand.thumbsup.fill")
+				.imageScale(.large)
+				.font(.title2)
+				.foregroundColor(.clear)
+				.read(maxImageHeightReader)
+			
+			main
+		}
+		.assignMaxPreference(for: maxImageHeightReader.key, to: $maxImageHeight)
+		.frame(maxHeight: .infinity)
+		.background(Color.primaryBackground)
+		.edgesIgnoringSafeArea([.bottom, .leading, .trailing]) // top is nav bar
+		.navigationBarTitle(
+			NSLocalizedString("lnurl-auth", comment: "Navigation bar title"),
+			displayMode: .inline
+		)
+		.zIndex(2) // [SendingView, ValidateView, LoginView, ScanView]
+	}
+	
+	@ViewBuilder
+	var main: some View {
+		
+		VStack(alignment: HorizontalAlignment.center, spacing: 30) {
+			
+			Spacer()
+			
+			Text("You can use your wallet to anonymously sign and authorize an action on:")
+				.multilineTextAlignment(.center)
+			
+			Text(domain())
+				.font(.headline)
+				.multilineTextAlignment(.center)
+			
+			if let model = mvi.model as? Scan.ModelLoginResult, model.error == nil {
+				
+				HStack(alignment: VerticalAlignment.firstTextBaseline) {
+					Image(systemName: "hand.thumbsup.fill")
+						.renderingMode(.template)
+						.imageScale(.large)
+						.frame(minHeight: maxImageHeight)
+					Text(successTitle())
+				}
+				.font(.title2)
+				.foregroundColor(Color.appPositive)
+				.padding(.top, 4)
+				.padding(.bottom, 5)
+				.padding([.leading, .trailing], 24)
+				
+			} else {
+				
+				Button {
```

Comment:
This button looks a bit strange, maybe larger than usual (for example compared to the pay button). Or is it just me ? 👀

Reply:
Done in 163dc42